arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -980,14 +1183,33 @@ public inline fun <A> Option<A>.orElse(alternative: () -> Option<A>): Option<A>
   return if (isEmpty()) alternative() else this
 }
 
+@Deprecated(
+  NicheAPI + "Prefer using the orElse method",
+  ReplaceWith("orElse(value)")
+)
 public infix fun <T> Option<T>.or(value: Option<T>): Option<T> = if (isEmpty()) {
   value
 } else {
   this
 }
 
+@Deprecated(
+  RedundantAPI + "Prefer using fromNullable",
+  ReplaceWith(
+    "Option.fromNullable(this)",
```

Comment:
Why is this recommended? we use toOption as the primary function for transforming nullable. This isn't consistent with the other recommendation for NonEmptyList where we recommend the extension function on List.

Reply:
I propose we keep this as well. It's a nice extension function which aligns well with the rest of arrow similarly to .some and .none.